dask/dataframe/groupby.py

```diff
@@ -3053,10 +3053,10 @@ def head(self, n=5, split_every=None, split_out=1, shuffle=None):
 
 
 def _unique_aggregate(series_gb, name=None):
-    ret = type(series_gb.obj)(
```

Comment:
I think you can do:

```
type(series_gb.obj)(
        [v.explode().unique() for _, v in series_gb], name=name, index=self.grouper.result_index
    )
```

for both groupby aggregations

Reply:
I think this way it looks a bit more readable.

Reply:
The reconstruction and astype would cause problems with MultiIndexes and EA Indexes (nullable and Arrow dtypes). Generally, we should try to rely on the pandas Index as much as possible

Reply:
Oh, the `index` part! thanks for the tip, I didn't see the change at first, and thought you're just talking about the list comprehension.

Reply:
Ok, so there is no `self` in this method, are you talking about `series_gb.obj`?

Reply:
Sorry, yes. Forgot to change it.

Reply:
Sorry, I misinterpreted the result and assumed the solution would be similar.

```
    return pd.concat(
        [v.groupby(by=series_gb.grouper.axis.names).sum() for _, v in
         series_gb],
    )
```

This is how pandas solves it internally and should translate to our problem (tests are passing)

Reply:
Elegant. ❤️ Thank you!

Reply:
@phofl | @j-bennet is `.grouper` a documented feature in pandas? Upon doing some digging it seems to be documented as an internal implementation of pandas, so this change is currently breaking `dask-cudf` because `GroupBy` in `cudf` doesn't have support for `.grouper` yet. We are evaluating on implementing `grouper` in `cudf` but wanted to first check if it is intended for usage as a public API or not?

Reply:
Tricky to draw the line here, ``SeriesGroupBy`` and ``DataFrameGroupBy`` are also considered more or less private-ish. 

We can look for another way to generate the index (not sure if there is an efficient one though).

Reply:
@galipremsagar could you open a separate issue for this?

Reply:
I opened a PR that reverts to the version without `grouper`. Definitely not as nice, but still fixes the bug.
https://github.com/dask/dask/pull/10182